### PR TITLE
Updates to AL-Go deployment process 

### PR DESCRIPTION
**Pull request by aholstrup1**
This PR changes the deployment process to
* Create pull requests when deploying Al-Go rather than force pushing 
* Fetch the OrgPat from a GitHub environment to ensure that the secret can only be accessed from workflows running on the main branch of AL-Go 
* (Update the OrgPat to be from a service account rather than a PAT from a Microsoft engineer)

New deployment process:
1. Run the Deploy workflow in microsoft/Al-Go
2. Request approval from another team member 
3. The deploy action will by default create PRs in Al-Go-PTE / Al-Go-AppSource / Al-Go-Actions. Go to those repositories, validate that the PR looks as expected and merge the new changes. 

As an example see:
https://github.com/aholstrup1/AL-Go/actions/runs/5853429237

https://github.com/aholstrup1/AL-Go-PTE/pull/19

### CODE REVIEW

File: .github/workflows/Deploy.yaml
Comment (mazhelez) [inline .github/workflows/Deploy.yaml:13]: Description should change as well then.

File: .github/workflows/Deploy.yaml
Comment (mazhelez) [inline .github/workflows/Deploy.yaml:18]: My take:
```suggestion
        description: Push directly to the target branch. If not set, a PR will be created.
```

File: .github/workflows/Deploy.yaml
Comment (mazhelez) [inline .github/workflows/Deploy.yaml:33]: Doesn't this change the game?

What if _Production_ environment does not exist? Do we need more docs?

File: Internal/Deploy.ps1
Comment (mazhelez) [inline Internal/Deploy.ps1:33]: Minor: my preference for uniqueness is to have a timestamp instead

```suggestion
        $branchName = "deploy/$baseBranch/$((Get-Date).ToUniversalTime().ToString(`"yyMMddHHmmss`"))"

```

File: Internal/Deploy.ps1
Comment (mazhelez) [inline Internal/Deploy.ps1:231]:
```suggestion
        PushChanges -BaseBranch $branch -CommitMessage "Deploying AL-Go from $algoBranch ($srcSHA) to $branch" -DirectCommit $directCommit

```

File: .github/workflows/Deploy.yaml
Comment (aholstrup1) [reply .github/workflows/Deploy.yaml:33]: I'll test this but no I don't believe you need the environment as long as you have an ORGPAT secret defined

Comment (aholstrup1) [reply .github/workflows/Deploy.yaml:33]: Tested it this morning. GitHub is still able to run the workflow as long as you have the OrgPat defined as a secret. As far as I can tell, the only change is that GitHub will automatically create a "Production" environment for you if it doesn't exist.

File: .github/workflows/Deploy.yaml
Comment (mazhelez) [inline .github/workflows/Deploy.yaml:71]: No need of the `else` statement on line 63.

**Review by mazhelez — APPROVED** [review]

File: Internal/Deploy.ps1
Comment (aholstrup1) [reply Internal/Deploy.ps1:30]: We will create a branch from origin/main if the base branch doesn't already exist. 
* If we want to deploy to PPPreview and the branch already exists, we won't need to create a new branch.
* If we want to deploy to PPPreview and the branch doesn't exist, then we will create it from main + Create a PR into that branch with the changes you're trying to deploy (The PPPreview changes coming from your Al-Go branch)

Comment (mazhelez) [reply Internal/Deploy.ps1:30]: Got it. New branches should always be based on _main_.

File: Internal/Deploy.ps1
Comment (mazhelez) [inline Internal/Deploy.ps1:33]:
```suggestion
        $branchName = "deploy/$BaseBranch/$((Get-Date).ToUniversalTime().ToString(`"yyMMddHHmmss`"))"

```

**Review by mazhelez — DISMISSED** [review]

File: .github/workflows/Deploy.yaml
Comment (freddydk) [inline .github/workflows/Deploy.yaml:33]: Why do we have an environment defined here?

Comment (freddydk) [reply .github/workflows/Deploy.yaml:33]: Read the comments in the description now

File: .github/workflows/Deploy.yaml
Comment (freddydk) [reply .github/workflows/Deploy.yaml:42]: read the description...

File: .github/workflows/Deploy.yaml
Comment (freddydk) [inline .github/workflows/Deploy.yaml:37]: Do we need code for this? Can't this approval just be setup on the environment?

File: Internal/Deploy.ps1
Comment (aholstrup1) [reply Internal/Deploy.ps1]: Add some help messages / comments

File: .github/workflows/Deploy.yaml
Comment (aholstrup1) [reply .github/workflows/Deploy.yaml:37]: We don't need it no :) And I'm open to removing it.

Without it, any of us would be able to Deploy, Approve and Merge changes to Al-Go preview / release branches on our own. 

GitHub does have an environment setting that lets you require approval from a list of approvers, but anyone in that list can also approve their own deployments. This might be okay though.. The "Validate Deployment" was just a way to force a "Sign-off" on the deployment from **someone else** before it deploys. 

Let me know what you think. I'm okay with either keeping it or removing it.

Comment (freddydk) [reply .github/workflows/Deploy.yaml:37]: Let's keep it in - but not require it when deploying to preview...

**Review by freddydk — APPROVED** [review]

**Pull request merged**
